src/worker/upstreamcommands.js

```diff
@@ -418,6 +418,16 @@ commands.NICK = async function(msg, con) {
         await con.messages.storeMessage(msg, con, null);
     }
 
+    // Update nick in buffer users
+    for (let bufferName in con.state.buffers) {
+        let buffer = con.state.buffers[bufferName];
+        let user = buffer.users[msg.nick.toLowerCase()];
+        if (user) {
+            buffer.removeUser(msg.nick);
+            buffer.addUser(msg.params[0], user);
```

Comment:
I'm thinking a `buffer.renameUser()` would be better to add? It would be easier to test if we ever get tests in place too